test/behaviour/reasoner/verification/CompletenessVerifier.java

```diff
@@ -73,7 +73,7 @@ private void verifyConjunction(BoundConjunction boundConjunction) {
                     verifyConclusionReasoning(materialisation.boundConclusion());
                 });
                 verifyConcludableReasoning(boundConcludable);
-            }
+            }  // TODO: If not inferred should we do a traversal to check?
```

Comment:
can we resolve this TODO?

Reply:
@jmsfltchr , 
I see no harm in doing it. We'd just be checking whether the reasoner returns retrieved answers as well.

Reply:
```
int numReasonedAnswersReceived = numReasonedAnswers(boundConcludable.pattern(), boundConcludable.concludable().retrieves());
                if (numReasonedAnswersReceived != 1) { /*throw*/ }
```
I thought this would work, but it seems to find 0 answers quite often. Am I doing something obviously wrong?

Reply:
Second thought: How would we do it? And is there a point to it? 
We'd be using traversal to fetch it both times. Maybe once we should do it with infer turned on and once with it turned off and see whether both get the same (0 or 1) number of results

Reply:
There's no such thing as inferred on or off for a traversal. Where we have non-inferred answers only coming from a traversal then we've implemented that by adding a filter.
Is there a point to it? I'm not sure there is. In doing so I think we might actually be testing `isInferredAnswer` more than anything else. I'm satisfied to skip it.

Reply:
If a thing can be both inferred and retrieved, wouldn' it pass through a ConcludableProcessor in case reasoning is enabled?
But my previous comment was wrong. To verify, Completeness verifier would just do a query (with infer on) and see whether the answer is being returned. It seems like a harmless check worth doing (and it uncovered a weird/unrelated bug in ForwardChainingMaterializer when I tried to do it)
So Joshua and I think we'll do it, but in a different PR since it's unrelated to checking explanation completeness.